### PR TITLE
Wr checker

### LINKED ISSUE #4: Record check command
A command for checking global records on the specified map. [SourceJump API](https://sourcejump.net/api) should be used for retrieving the record and other information.

Current implementation just listen to messages sent in a channel but a slash command (`/record {map}`) might be a better solution for this?

record

### PR DESCRIPTION
**Pull request by pxbhop**
Created the WR checker command, tried to mimic Tony's current embed but couldn't quite get it exactly the same. If you want to mess with it to get it closer to the original feel free.
image
Command uses Discord's built in Slash Command feature. If a map isn't found in the SJ database, the response is ephemeral (can only be seen by the person that sent the command). This is to prevent pings being sent from that message.

### CODE REVIEW

**Comment by log-ical** [comment]
https://github.com/jelsum/discord-bot/pull/1 
Make sure we update the env example for the API key

**Comment by log-ical** [comment]
https://github.com/jelsum/discord-bot/pull/2
> If you want to mess with it to get it closer to the original feel free.

Made the formatting to match the Sourcejump version.

(Excuse the weird numbers and missing information, didn't have an API key so I had to improvise some numbers when testing.)
image

**Comment by ecsr** [comment]
Will look at this once https://github.com/jelsum/discord-bot/pull/1 and https://github.com/jelsum/discord-bot/pull/2 has been merged.

File: src/commands/wr.js [hunk 1]
```diff
@@ -0,0 +1,112 @@
+require('dotenv').config();
```
Comment (ecsr) [inline src/commands/wr.js]: Pretty sure this is not needed as the .env file already gets loaded first thing in index.js so process.env should already be populated with our variables.

File: src/commands/wr.js [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+        )
+        .addStringOption((option) =>
+            option
+                .setName('map')
+                .setDescription('Name of the map to check')
+                .setRequired(true),
+        ),
+    async execute(interaction) {
+        // Maybe store this in env so it doesn't need to be changed/declared here?
+        const SOURCEJUMP_API_URL = 'https://sourcejump.net/api';
```
Comment (ecsr) [inline src/commands/wr.js]: Lets store this in our .env file and rename SJ_API_KEY to SOURCEJUMP_API_KEY.

File: src/commands/wr.js [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+        const map = interaction.options.getString('map');
+        const apiOptions = {
+            method: 'GET',
+            headers: {
+                'api-key': process.env.SJ_API_KEY,
+            },
+        };
+
+        fetch(`${SOURCEJUMP_API_URL}/records/${map}`, apiOptions)
+            .then((response) => response.text())
```
Comment (ecsr) [inline src/commands/wr.js]: Maybe change this to response.json()?

File: src/commands/wr.js [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+            .then((body) => {
+                if (body.length === 2) {
```
Comment (ecsr) [inline src/commands/wr.js]: This would need to be changed to `body.length === 0` if `text()` is changed to `json()`.

File: src/commands/wr.js [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+                    return interaction.reply({
+                        content: `No times found for ${map}.`,
+                        ephemeral: true,
+                    });
+                }
+
+                body = JSON.parse(body);
```
Comment (ecsr) [inline src/commands/wr.js]: And this can be removed.

**Review by ecsr — APPROVED** [review]

**Comment by ecsr** [comment]
Thanks @jelsum and @log-ical.

**Pull request merged**